src/models/orders.rs

```diff
@@ -53,13 +45,9 @@ impl Order {
     #[allow(dead_code)]
     pub fn validate_order(&self) -> Result<bool> {
         let message = self.get_digest()?;
-        let signature = Signature {
-            v: self.signature_v - 27 as u8,
-            r: *self.signature_r.as_fixed_bytes(),
-            s: *self.signature_s.as_fixed_bytes(),
-        };
-        let owner = signature.recover(&message)?;
-        return Ok(H160::from(owner.address()).eq(&self.owner));
+        let signature = encode(&[self.signature_r.into_token(), self.signature_s.into_token()]);
```

Comment:
`web3` provides a cool tool for building the signature for you <https://docs.rs/web3/0.13.0/web3/types/struct.Recovery.html>. You should be able to do:
```rust
let signature = Recovery::new(&message, self.signature_v, self.signature_r, self.signature_s).as_signature();
```

This way you don't have to worry about the details of encoding the r and s words, or converting the v-value into a recovery ID (from Electrum notation, or dealing with chain relay protection).